modules/ppcp-settings/resources/js/Components/ReusableComponents/Controls/SoftdescriptorInput.js

```diff
@@ -0,0 +1,19 @@
+import ControlTextInput from './ControlTextInput';
+
+const SoftDescriptorInput = ( { value, onChange, placeholder } ) => {
+	const handleChange = ( newValue ) => {
+		if ( newValue.length <= 22 ) {
+			onChange( newValue );
+		}
+	};
```

Comment:
As this is a unique field, I think it's not too useful to create a reusable component for it.

Suggestion 1: Move this method into the `PaypalSettings.js` file instead of creating a new component.
Suggestion 2: Wrap the method in a `useCallback()` hook.

Why useCallback:
This makes the function "stable", which means, the function is cached by React, and does not change when any of the other props change. In this specific case, it has very little impact, but it's a good practice to get used to, as it avoids unnecessarily re-rendering components.

```js
const handleChange = useCallback(( newValue ) => {
	if ( newValue.length <= 22 ) {
		onChange( newValue );
	}
}, [ onChange ]); // ← only if "onChange" prop changes, a new copy of the "handleChange" function is generated
```